airflow/providers/databricks/hooks/databricks_sql.py

```diff
@@ -80,8 +85,18 @@ def __init__(
         self.http_headers = http_headers
         self.catalog = catalog
         self.schema = schema
+        self.return_serializable = return_serializable
         self.additional_params = kwargs
 
+        if not self.return_serializable:
```

Comment:
Who will set this and why and when? Is this a kind of backwards compatibility? Will this not generate a lot of warnings?

Reply:
Yes, this is to ensure the change is backward compatible.

Users of the hook will be warned. Users of the `DatabricksSqlOperator` won't, because the operator was already returning a list of list - which becomes a list of namedtuple with this PR. Thus, it will generate noise only to hook users.

The goal is to effectively do a breaking change later. But if we don't want that - or not _that_ soon, I can check to print the warning when the provider loads for now ?

Reply:
Will clarify the docstring and the warning